### ISSUE TITLE
No call popups / inbound and outbound calls are recorded

### ISSUE THREAD

**Issue by bludauma** [blak3r/yaai#97]
Hey,
i am very happy to manage inbound and outbound calls with your tool.
Inbound and ountbound calls are well redcorded in sugar and click to dial is working.

But there are no popups, neither for inbound or outbound calls.

Here is the output when i start asteriskLogger.php:

[Config Settings]
  Sugar root set to [/var/www/sugarcrm/custom/modules/Asterisk/../../../]
  Logging to: /var/www/sugarcrm/custom/modules/Asterisk/logger_log
  Callout prefix is []
  Callin prefix is []
  Match Internal Regex = /^(sip\/[1-9][0-9][0-9][0-9][0-9][0-9]-)/i
[Database Connection]
  Selecting DB Name: sugar
[SOAP Connection]
  Trying SOAP login endpoint=[http://...../sugarcrm/soap.php] user=[Asterisk] password=[b96a150555f019811198a2dd13a15ac1]
PHP Warning:  Attempt to modify property of non-object in /var/www/sugarcrm/include/nusoap/nusoap.php on line 6171
PHP Warning:  Attempt to modify property of non-object in /var/www/sugarcrm/include/nusoap/nusoap.php on line 6171
PHP Warning:  Attempt to modify property of non-object in /var/www/sugarcrm/include/nusoap/nusoap.php on line 6171
PHP Warning:  Attempt to modify property of non-object in /var/www/sugarcrm/include/nusoap/nusoap.php on line 6171
  Successfully logged into Sugar via SOAP!  SessionId=8uf0vifsisqmd4igp9oi681865 user=Asterisk GUID=7f100256-54eb-3277-c6a2-507e8fb4aac9
[Asterisk Manager Interface (AMI) Connection]
  Successfully opened socket connection to ......:5038
  AMI Version Info:
    `
    Asterisk Call Manager/1.1
`
  Supported AMI version: 1.1 Detected
  AMI Login action raw response:
    `
    Response: Success
    Message: Authentication accepted
`
  AMI Login was a _success!_
Waiting for call events...
.......

I also checked the userights in manager.conf and switched them to read: all and write: all, as described in the trouble shooting section.
Furthermore calls produce proper events on my asterisk.

Best regard, Malte

**Comment by blak3r** [comment]
If both inbound and outbound calls are being logged in Calls, but your just not getting callPopups it's not an AMI issue. 

Are the calls being assigned to the correct user? Ie the one that matches their extension. 

Also check the wiki page on chrome debugging to see if you have any errors. 

Finally, you have call notification checkbox added in studio and checked?

~blake

**Comment by bludauma** [comment]
Yes the calls are assigned to the correct user. I don't use chrome (i use firefox or ie) and the checkbox "call notifications" is enabled.
Furthermore there are no javascrip or php errors on my sugar page.

**Comment by blak3r** [comment]
Ok.  Well check to see if you have any JavaScript errors or errors in php. 

Google how to see if there are any JavaScript errors in the browser of your choice. 

If no errors there, then look in your php log. 

What version of Sugarcrm are you using?

**Comment by jernejp** [comment]
Hi! I have the same issue. Click to dial works, inbound calls get logged and linked to contacts. Everything seem to work, but popups. Installed on Sugar 6.5.8 and Asterisk 1.6.2.9-2+squeeze8, both running on Debian squeeze. Tried with Chrome, Firefox, Safari and Opera.

**Comment by blak3r** [comment]
CE version or Pro?

Any errors in JavaScript logs? 

Do you see requests going back and forth in the networking tab in chrome. If so, is server always returning [.]?

~blake

**Comment by jernejp** [comment]
CE version. In networking tab in chrome I managed to track down an Internal Server Error 500 in http://domain.com/sugarcrm/index.php?entryPoint=AsteriskCallListener - initiator:  http://domain.com/sugarcrm/cache/include/javascript/sugar_grp1_jquery.js?v=KIdymiwc0f9SM6pyHh567A
I think that could be the cause of the problem...

**Comment by blak3r** [comment]
Yeah that's your issue.  So, check your php logs now and see what the error is there.  You should have a php error if you're getting a 500.

**Comment by jernejp** [comment]
It turns out it was a missing language file  -> PHP Fatal error:  require(): Failed opening required 'custom/modules/Asterisk/language/sl_si.lang.php'. I just copied en_us.lang.php to sl_si.lang.php and now it works. I will translate the module to slovene language and send it to you. Thank you! :)

**Comment by blak3r** [comment]
Ahhh! Good find.  Yeah I'd very much appreciate the language file.
Any chance you could write up a step by step guide for how to update the
language files?

**Comment by Tigger28** [comment]
Same Problem here with the German file ge_ge.lang. Looks like that the file is not complete. I miss the part 

...

'HANGUP'                =>  'Finished',

```
// Added in v2.2
'ASTERISKLBL_DURATION'         => 'Duration',
'ASTERISKLBL_SELECTCONTACT'    => 'Select Contact',
'ASTERISKLBL_MULTIPLE_MATCHES' => "Multiple Matches",
'ASTERISKLBL_CALLERID'         => 'CallerID',

// For asteriskLogger.php SOAP call entries
'CALL_AUTOMATIC_RECORD'         => '** Automatic record **',
'CALL_IN_LIMBO'                 => 'In Limbo',
'CALL_STATUS_HELD'              => 'Held',
'CALL_STATUS_MISSED'            => 'Missed',
'CALL_NAME_CALL'                => 'CALL',
'CALL_NAME_MISSED'              => 'Missed Call',
'CALL_DESCRIPTION_CALLER_ID'    => 'Caller ID',
'CALL_DESCRIPTION_MISSED'       => 'Missed/failed call',
'CALL_DESCRIPTION_PHONE_NUMBER' => 'Phone Number'
```

   );

?>

Dont know why didn't play around with this file. Maybe its already missing since the installation ;-)

**Comment by blak3r** [comment]
Yeah I don't maintain the language files.  They are from a time long long ago.  We've added lots of new properties since then.

This just reminded me... getting language files working is quite a hassle.  Once you translate them I believe you have to reinstall yaai in order for it to take affect.  We might also have to update the manifest to include the language files.  

Here's a pull request @trustmaster did for Russian language support... If i recall we already had russian language support in the manifest.  I know SL isn't already in the manifest so we probably have to add entries there as well.  Once the changes are done, you'll have to zip up the folder and install through module loader.

@trustmaster - you've done more localization then I have.  Is my assessment correct?

**Comment by trustmaster** [comment]
Right, the manifest should be updated if you add new localization files. Reinstall is not required if you put them in the right place by hand. Updating existing language files doesn't require any extra actions.

**Comment by blak3r** [comment]
Thanks @trustmaster 

I created a wiki page which we should get a write up together for how to translate yaai.  https://github.com/blak3r/yaai/wiki/How-to-Translate

Anyone can edit the wiki pages so if you want to take a stab @Tigger28 that'd be great.

**Comment by Tigger28** [comment]
I have a Eng, Russian and German Language file. Eng and Russion works really well. Only the German file seems to be broken. I updated the file an put it at the right location by hand, but it didnt take any affect, No Popups :-(
I try to find a solution today.

**Comment by Tigger28** [comment]
There we go :-)
It finally works, found the Problem.
1. the language file ge_ge.lang was not complete ( look at my comment above ) 
2. couple days ago i installed a new german language file in my SugarCRM CE 6.5.8 this file is named de_de.lang, so i renamed my language file in "custom/modules/Asterisk/language/" to de_de.lang finally the Popups show up.

**Comment by Tigger28** [comment]
I made a comment in Wiki you may correct me because my Eng isnt the best :-)
